src/axom/sidre/tests/sidre_mfem_datacollection.cpp

```diff
@@ -261,6 +259,197 @@ TEST(sidre_datacollection, dc_alloc_nonowning_parmesh)
+                     non_const_parmesh.GroupNEdges(i),
+                     non_const_parmesh.GroupNTriangles(i),
+                     non_const_parmesh.GroupNQuadrilaterals(i)};
+  }
+
+  return result;
+}
+
+/**
+ * @brief Helper method for testing that a parallel mesh is reconstructed correctly
+ * @param [in] base_mesh The serial mesh object to distribute, save, and then reload
+ * @param [in] part_method The partitioning method to use - in [0, 5]
+ */
+static void testParallelMeshReload(mfem::Mesh& base_mesh,
+                                   const int part_method = 1)
+{
+  mfem::ParMesh parmesh(MPI_COMM_WORLD, base_mesh, nullptr, part_method);
+
+  mfem::H1_FECollection fec(1, base_mesh.Dimension());
+  mfem::ParFiniteElementSpace parfes(&parmesh, &fec);
+
+  // The mesh must be owned by Sidre to properly manage data in case of
+  // a simulated restart (save -> load)
+  const bool owns_mesh = true;
+  MFEMSidreDataCollection sdc_writer(testName(), &parmesh, owns_mesh);
+
+  // Save some basic info about the mesh
+  const int n_verts = sdc_writer.GetMesh()->GetNV();
+  const int n_ele = sdc_writer.GetMesh()->GetNE();
+  const int n_bdr_ele = sdc_writer.GetMesh()->GetNBE();
+
+  // ParMesh-specific info
+  auto writer_pmesh = dynamic_cast<mfem::ParMesh*>(sdc_writer.GetMesh());
+  ASSERT_NE(writer_pmesh, nullptr);
+  const int n_groups = writer_pmesh->GetNGroups();
+  const int n_face_neighbors = writer_pmesh->GetNFaceNeighbors();
+  const int n_shared_faces = writer_pmesh->GetNSharedFaces();
+
+  // Group-specific info
+  auto writer_group_data = getGroupData(*writer_pmesh);
+
+  sdc_writer.SetCycle(0);
+  sdc_writer.Save();
+
+  MFEMSidreDataCollection sdc_reader(testName());
+
+  // Needs to be set "manually" in order for everything to be loaded in properly
+  sdc_reader.SetComm(MPI_COMM_WORLD);
+  sdc_reader.Load();
+
+  // Make sure the mesh was actually reconstructed
+  EXPECT_EQ(sdc_reader.GetMesh()->GetNV(), n_verts);
+  EXPECT_EQ(sdc_reader.GetMesh()->GetNE(), n_ele);
+  EXPECT_EQ(sdc_reader.GetMesh()->GetNBE(), n_bdr_ele);
+
+  // Make sure the ParMesh was reconstructed - even on one rank, this
+  // will still be a ParMesh
+  auto reader_pmesh = dynamic_cast<mfem::ParMesh*>(sdc_reader.GetMesh());
+  ASSERT_NE(reader_pmesh, nullptr);
+  EXPECT_EQ(reader_pmesh->GetNGroups(), n_groups);
+  EXPECT_EQ(reader_pmesh->GetNFaceNeighbors(), n_face_neighbors);
+  EXPECT_EQ(reader_pmesh->GetNSharedFaces(), n_shared_faces);
+
+  auto reader_group_data = getGroupData(*reader_pmesh);
+  EXPECT_EQ(writer_group_data, reader_group_data);
+  EXPECT_TRUE(sdc_reader.verifyMeshBlueprint());
+}
+
+/**
+ * @brief Wrapper for testing mesh reconstruction across all partitioning methods
+ * @param [in] base_mesh The serial mesh object to distribute, save, and then reload
+ */
+static void testParallelMeshReloadAllPartitionings(mfem::Mesh& base_mesh)
+{
+  for(int part_method = 0; part_method <= 5; part_method++)
```

Comment:
Magic number:  Where does `5` come from?  Perhaps it would be clearer to use a named constant instead?

Reply:
That number isn't documented anywhere in MFEM to my knowledge (determined through source code inspection), but I've replaced it with a named constant in these tests.